src/systems/environment_preload/EnvironmentPreload.cc

```diff
@@ -160,26 +209,140 @@ void EnvironmentPreload::PreUpdate(
       }
     }
 
+    this->dataDescription.set_static_time(ignoreTime);
+    this->dataDescription.set_coordinate_type(spatialReference);
+    this->dataDescription.set_time(timeColumnName);
+    this->dataDescription.set_x(spatialColumnNames[0]);
+    this->dataDescription.set_y(spatialColumnNames[1]);
+    this->dataDescription.set_z(spatialColumnNames[2]);
+
+    needsReload = true;
+  }
+
+  //////////////////////////////////////////////////
+  public: components::EnvironmentalData::ReferenceUnits ConvertUnits(
+    const Units &_unit)
+  {
+    switch (_unit) {
+      case Units::DataLoadPathOptions_DataAngularUnits_DEGREES:
+        return components::EnvironmentalData::ReferenceUnits::DEGREES;
+      case Units::DataLoadPathOptions_DataAngularUnits_RADIANS:
+        return components::EnvironmentalData::ReferenceUnits::RADIANS;
+      default:
+        gzerr << "Invalid unit conversion" << std::endl;
```

Comment:
Maybe this should additionally say it's assuming converting to radians, in case the user passes in something invalid that they expect to be degrees for some reason.